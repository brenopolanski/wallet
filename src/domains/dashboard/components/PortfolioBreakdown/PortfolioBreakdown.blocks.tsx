import React from "react";
import { useTranslation } from "react-i18next";

interface MoreDetailsButtonProperties {
	onClick: () => void;
}

const MoreDetailsButton: React.VFC<MoreDetailsButtonProperties> = ({ onClick }) => {
	const { t } = useTranslation();

	return (
		<div className="border-l border-theme-secondary-300 dark:border-theme-secondary-800 ml-4 pl-4">
			<button onClick={onClick} type="button" className="font-semibold link">
				{t("COMMON.MORE_DETAILS")}
			</button>
		</div>
	);
};

interface LabelledTextProperties {
	label: string;
	children: (textClassName: string) => JSX.Element;
}

const LabelledText: React.FC<LabelledTextProperties> = ({ label, children }) => (
	<div className="flex flex-col font-semibold space-y-1 mx-2">
		<span className="text-sm text-theme-secondary-500 dark:text-theme-secondary-700">{label}</span>

		{children("text-lg text-theme-secondary-900 dark:text-theme-secondary-200")}
	</div>
);

export { LabelledText, MoreDetailsButton };
