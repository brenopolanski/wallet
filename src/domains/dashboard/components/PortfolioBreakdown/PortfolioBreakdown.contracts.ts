import { LineGraphDataPoint } from "app/components/Graphs/LineGraph/LineGraph.contracts";

interface LegendProperties {
	dataPoints: LineGraphDataPoint[];
	hasZeroBalance: boolean;
	onMoreDetailsClick: () => void;
}

interface LabelledTextProperties {
	label: string;
	children: (textClassName: string) => JSX.Element;
}

interface TooltipProperties {
	dataPoint: LineGraphDataPoint;
}

export type { LabelledTextProperties, LegendProperties, TooltipProperties };
