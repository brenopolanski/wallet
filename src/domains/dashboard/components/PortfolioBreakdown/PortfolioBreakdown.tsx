import { sortByDesc } from "@payvo/sdk-helpers";
import { Contracts, Helpers } from "@payvo/sdk-profiles";
import { Amount } from "app/components/Amount";
import { EmptyBlock } from "app/components/EmptyBlock";
import { GraphDataPoint } from "app/components/Graphs/Graphs.contracts";
import { GRAPH_COLOR_EMPTY, GRAPH_COLOR_EMPTY_DARK } from "app/components/Graphs/Graphs.shared";
import { LineGraph } from "app/components/Graphs/LineGraph";
import { useTheme } from "app/hooks";
import { PortfolioBreakdownDetails } from "domains/dashboard/components/PortfolioBreakdownDetails";
import { usePortfolioBreakdown } from "domains/dashboard/hooks/use-portfolio-breakdown";
import React, { useMemo, useState } from "react";
import { Trans, useTranslation } from "react-i18next";

import { LabelledText, Legend, PortfolioBreakdownSkeleton, Tooltip } from "./PortfolioBreakdown.blocks";
import { formatPercentage, getColor } from "./PortfolioBreakdown.helpers";

interface PortfolioBreakdownProperties {
	profile: Contracts.IProfile;
	profileIsSyncingExchangeRates: boolean;
}

export const PortfolioBreakdown: React.VFC<PortfolioBreakdownProperties> = ({
	profile,
	profileIsSyncingExchangeRates,
}) => {
	const { t } = useTranslation();
	const { isDarkMode } = useTheme();

	const [isDetailsOpen, setIsDetailsOpen] = useState(false);

	const { loading, balance, assets, walletsCount, ticker } = usePortfolioBreakdown({
		profile,
		profileIsSyncingExchangeRates,
	});

	const hasZeroBalance = useMemo(() => balance === 0, [balance]);

	const lineGraphData = useMemo<GraphDataPoint[]>(() => {
		if (hasZeroBalance) {
			return assets.map((asset) => ({
				color: isDarkMode ? GRAPH_COLOR_EMPTY_DARK : GRAPH_COLOR_EMPTY,
				data: {
					amountFormatted: Helpers.Currency.format(0, ticker),
					label: asset.label,
					percentFormatted: `0%`,
				},
				value: 0,
			}));
		}

		return sortByDesc(assets, "percent").map((asset, index) => ({
			color: getColor(index, isDarkMode),
			data: {
				amountFormatted: Helpers.Currency.format(asset.convertedAmount, ticker),
				label: asset.label,
				percentFormatted: formatPercentage(asset.percent),
			},
			value: asset.percent,
		}));
	}, [assets, hasZeroBalance, isDarkMode, ticker]);

	if (loading && assets.length === 0) {
		return <PortfolioBreakdownSkeleton />;
	}

	if (assets.length === 0) {
		return (
			<EmptyBlock>
				<Trans i18nKey="DASHBOARD.PORTFOLIO_BREAKDOWN.EMPTY" components={{ bold: <strong /> }} />
			</EmptyBlock>
		);
	}

	return (
		<>
			<div
				className="py-4 px-6 bg-theme-secondary-100 dark:bg-black rounded-xl flex"
				data-testid="PortfolioBreakdown"
			>
				<div className="flex space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800">
					<LabelledText label={t("COMMON.YOUR_BALANCE")}>
						{(textClassName) => <Amount className={textClassName} ticker={ticker} value={balance} />}
					</LabelledText>

					<LabelledText label={t("COMMON.ASSETS")}>
						{(textClassName) => (
							<span data-testid="PortfolioBreakdown__assets" className={textClassName}>
								{assets.length}
							</span>
						)}
					</LabelledText>

					<LabelledText label={t("COMMON.WALLETS")}>
						{(textClassName) => (
							<span data-testid="PortfolioBreakdown__wallets" className={textClassName}>
								{walletsCount}
							</span>
						)}
					</LabelledText>
				</div>

				<div className="flex-1 ml-6">
					<LineGraph
						data={lineGraphData}
						renderAsEmpty={hasZeroBalance}
						renderTooltip={(dataPoint) => <Tooltip dataPoint={dataPoint} />}
						renderLegend={(dataPoints) => (
							<Legend
								dataPoints={dataPoints}
								hasZeroBalance={hasZeroBalance}
								onMoreDetailsClick={() => setIsDetailsOpen(true)}
							/>
						)}
					/>
				</div>
			</div>

			<PortfolioBreakdownDetails
				isOpen={isDetailsOpen}
				assets={assets}
				balance={balance}
				exchangeCurrency={ticker}
				onClose={() => setIsDetailsOpen(false)}
			/>
		</>
	);
};
