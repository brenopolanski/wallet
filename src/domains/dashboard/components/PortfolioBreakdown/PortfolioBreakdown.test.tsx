import { LSK } from "@payvo/sdk-lsk";
import { Contracts } from "@payvo/sdk-profiles";
import userEvent from "@testing-library/user-event";
import * as sharedGraphUtils from "app/components/Graphs/Graphs.shared";
import * as useThemeHook from "app/hooks/use-theme";
import { buildTranslations } from "app/i18n/helpers";
import nock from "nock";
import React from "react";
import { env, render, screen } from "utils/testing-library";

import { PortfolioBreakdown } from "./PortfolioBreakdown";

const translations = buildTranslations();

describe("PortfolioBreakdown", () => {
	let profile: Contracts.IProfile;

	let arkWallet: Contracts.IReadWriteWallet;
	let lskWallet: Contracts.IReadWriteWallet;

	let portfolioBreakdownMock: jest.SpyInstance;
	let isRestoredMock: jest.SpyInstance;
	let balanceMock: jest.SpyInstance;

	let useGraphWidthMock: jest.SpyInstance;

	beforeAll(async () => {
		nock.disableNetConnect();

		env.registerCoin("LSK", LSK);

		profile = env.profiles().create("blank");

		profile.settings().set(Contracts.ProfileSetting.ExchangeCurrency, "USD");

		arkWallet = (await profile.walletFactory().generate({ coin: "ARK", network: "ark.mainnet" })).wallet;
		lskWallet = (await profile.walletFactory().generate({ coin: "LSK", network: "lsk.mainnet" })).wallet;

		profile.wallets().push(arkWallet);
		profile.wallets().push(lskWallet);

		// Mock graph width to 100.
		useGraphWidthMock = jest.spyOn(sharedGraphUtils, "useGraphWidth").mockReturnValue([undefined as never, 100]);
	});

	afterAll(() => {
		nock.enableNetConnect();

		env.profiles().forget(profile.id());

		useGraphWidthMock.mockRestore();
	});

	beforeEach(() => {
		portfolioBreakdownMock = jest.spyOn(profile.portfolio(), "breakdown").mockReturnValue([
			{ coin: arkWallet.coin(), shares: 85, source: 85, target: 85 },
			{ coin: lskWallet.coin(), shares: 15, source: 15, target: 15 },
		]);

		isRestoredMock = jest.spyOn(profile.status(), "isRestored").mockReturnValue(true);
		balanceMock = jest.spyOn(profile, "convertedBalance").mockReturnValue(100);
	});

	afterEach(() => {
		portfolioBreakdownMock.mockRestore();
		isRestoredMock.mockRestore();
		balanceMock.mockRestore();
	});

	it.each([true, false])("should render with dark mode = %s", (isDarkMode) => {
		const useThemeMock = jest.spyOn(useThemeHook, "useTheme").mockReturnValue({ isDarkMode } as never);

		const { asFragment } = render(<PortfolioBreakdown profile={profile} profileIsSyncingExchangeRates={false} />);

		expect(screen.getByTestId("PortfolioBreakdown")).toBeInTheDocument();

		expect(screen.getByTestId("Amount")).toHaveTextContent("$100.00");
		expect(screen.getByTestId("PortfolioBreakdown__assets")).toHaveTextContent("2");
		expect(screen.getByTestId("PortfolioBreakdown__wallets")).toHaveTextContent("2");

		expect(screen.getByTestId("LineGraph__svg")).toBeInTheDocument();
		expect(screen.getAllByTestId("LineGraph__item")).toHaveLength(2);

		expect(screen.getByText(translations.COMMON.MORE_DETAILS)).not.toBeDisabled();

		expect(asFragment()).toMatchSnapshot();

		useThemeMock.mockRestore();
	});

	it("should render loading", () => {
		portfolioBreakdownMock.mockReturnValue([]);

		const { asFragment, rerender } = render(
			<PortfolioBreakdown profile={profile} profileIsSyncingExchangeRates={true} />,
		);

		expect(screen.getByTestId("PortfolioBreakdownSkeleton")).toBeInTheDocument();

		isRestoredMock.mockReturnValue(false);

		rerender(<PortfolioBreakdown profile={profile} profileIsSyncingExchangeRates={false} />);

		expect(screen.getByTestId("PortfolioBreakdownSkeleton")).toBeInTheDocument();

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render empty", () => {
		portfolioBreakdownMock.mockReturnValue([]);

		const { asFragment } = render(<PortfolioBreakdown profile={profile} profileIsSyncingExchangeRates={false} />);

		expect(screen.getByTestId("EmptyBlock")).toBeInTheDocument();
		expect(screen.getByTestId("EmptyBlock")).toHaveTextContent(/Your portfolio is currently empty/);

		expect(asFragment()).toMatchSnapshot();
	});

	it.each([true, false])("should render zero balance state with dark mode = %s", (isDarkMode) => {
		const useThemeMock = jest.spyOn(useThemeHook, "useTheme").mockReturnValue({ isDarkMode } as never);

		balanceMock.mockReturnValue(0);

		const { asFragment } = render(<PortfolioBreakdown profile={profile} profileIsSyncingExchangeRates={false} />);

		expect(screen.getByTestId("PortfolioBreakdown")).toBeInTheDocument();
		expect(screen.getByTestId("LineGraph__empty")).toBeInTheDocument();

		expect(screen.getByText(translations.COMMON.MORE_DETAILS)).toBeDisabled();

		expect(asFragment()).toMatchSnapshot();

		useThemeMock.mockRestore();
	});

	it("should have a button to open detail modal", () => {
		render(<PortfolioBreakdown profile={profile} profileIsSyncingExchangeRates={false} />);

		expect(screen.getByText(translations.COMMON.MORE_DETAILS)).toBeEnabled();

		userEvent.click(screen.getByText(translations.COMMON.MORE_DETAILS));

		expect(screen.getByTestId("modal__inner")).toBeInTheDocument();
		expect(screen.getByText(translations.DASHBOARD.PORTFOLIO_BREAKDOWN_DETAILS.TITLE)).toBeInTheDocument();

		userEvent.click(screen.getByTestId("modal__close-btn"));

		expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument();
	});

	it("should show tooltip when hovering graph elements", () => {
		render(<PortfolioBreakdown profile={profile} profileIsSyncingExchangeRates={false} />);

		expect(screen.getByTestId("LineGraph__svg")).toBeInTheDocument();
		expect(screen.getAllByTestId("LineGraph__item")).toHaveLength(2);

		expect(screen.queryByTestId("PortfolioBreakdown__tooltip")).not.toBeInTheDocument();

		userEvent.hover(screen.getAllByTestId("LineGraph__item")[0]);

		expect(screen.getByTestId("PortfolioBreakdown__tooltip")).toBeInTheDocument();
		expect(screen.getByTestId("PortfolioBreakdown__tooltip")).toHaveTextContent(/ARK/);
		expect(screen.getByTestId("PortfolioBreakdown__tooltip")).toHaveTextContent(/\$85.00/);
		expect(screen.getByTestId("PortfolioBreakdown__tooltip")).toHaveTextContent(/85%/);

		userEvent.unhover(screen.getAllByTestId("LineGraph__item")[0]);
		userEvent.hover(screen.getAllByTestId("LineGraph__item")[1]);

		expect(screen.getByTestId("PortfolioBreakdown__tooltip")).toHaveTextContent(/LSK/);
		expect(screen.getByTestId("PortfolioBreakdown__tooltip")).toHaveTextContent(/\$15.00/);
		expect(screen.getByTestId("PortfolioBreakdown__tooltip")).toHaveTextContent(/15%/);
	});
});
