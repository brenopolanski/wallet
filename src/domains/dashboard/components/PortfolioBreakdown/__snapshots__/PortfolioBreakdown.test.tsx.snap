// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioBreakdown should render empty 1`] = `
<DocumentFragment>
  <div
    class="border border-solid border-theme-secondary-300 dark:border-theme-secondary-800 rounded-lg text-theme-secondary-text text-center leading-5 p-6"
    data-testid="EmptyBlock"
  >
    <div>
      Your portfolio is currently empty. 
      <strong>
        Create
      </strong>
       or 
      <strong>
        Import
      </strong>
       a wallet to get started.
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioBreakdown should render loading 1`] = `
<DocumentFragment>
  <div
    class="py-4 px-6 bg-theme-secondary-100 dark:bg-black rounded-xl flex"
    data-testid="PortfolioBreakdownSkeleton"
  >
    <div
      class="flex space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800"
    >
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Your Balance
        </span>
        <div
          class="flex items-center h-7"
        >
          <span>
            <span
              class="flex items-center leading-none"
            >
              <span
                class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                style="width: 100px; height: 18px;"
              >
                ‌
              </span>
              ‌
            </span>
          </span>
        </div>
      </div>
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Assets
        </span>
        <div
          class="flex items-center h-7"
        >
          <span>
            <span
              class="flex items-center leading-none"
            >
              <span
                class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                style="width: 20px; height: 18px;"
              >
                ‌
              </span>
              ‌
            </span>
          </span>
        </div>
      </div>
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Wallets
        </span>
        <div
          class="flex items-center h-7"
        >
          <span>
            <span
              class="flex items-center leading-none"
            >
              <span
                class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                style="width: 20px; height: 18px;"
              >
                ‌
              </span>
              ‌
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex-1 ml-4 self-end"
    >
      <span>
        <span
          class="flex items-center leading-none"
        >
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style="width: 0px; height: 8px;"
          >
            ‌
          </span>
          ‌
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioBreakdown should render with dark mode = false 1`] = `
<DocumentFragment>
  <div
    class="py-4 px-6 bg-theme-secondary-100 dark:bg-black rounded-xl flex"
    data-testid="PortfolioBreakdown"
  >
    <div
      class="flex space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800"
    >
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Your Balance
        </span>
        <span
          class="whitespace-nowrap text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="Amount"
        >
          $100.00
        </span>
      </div>
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Assets
        </span>
        <span
          class="text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="PortfolioBreakdown__assets"
        >
          2
        </span>
      </div>
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Wallets
        </span>
        <span
          class="text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="PortfolioBreakdown__wallets"
        >
          2
        </span>
      </div>
    </div>
    <div
      class="flex-1 ml-6"
    >
      <div>
        <div
          class="absolute duration-200 transition-opacity opacity-0"
        />
        <div
          class="flex justify-end mb-1"
          data-testid="LineGraph__legend"
        >
          <div
            class="flex justify-end mb-1"
          >
            <div
              class="flex space-x-4"
            >
              <div
                class="flex items-center space-x-1 text-sm font-semibold"
              >
                <span
                  class="h-3 w-1 rounded bg-theme-success-600"
                />
                <span
                  class="text-theme-secondary-700 dark:text-theme-secondary-200"
                >
                  ARK
                </span>
                <span
                  class="text-theme-secondary-500 dark:text-theme-secondary-700"
                >
                  85%
                </span>
              </div>
              <div
                class="flex items-center space-x-1 text-sm font-semibold"
              >
                <span
                  class="h-3 w-1 rounded bg-theme-warning-600"
                />
                <span
                  class="text-theme-secondary-700 dark:text-theme-secondary-200"
                >
                  LSK
                </span>
                <span
                  class="text-theme-secondary-500 dark:text-theme-secondary-700"
                >
                  15%
                </span>
              </div>
            </div>
            <div
              class="border-l border-theme-secondary-300 dark:border-theme-secondary-800 ml-4 pl-4"
            >
              <button
                class="font-semibold text-sm link"
                type="button"
              >
                More Details
              </button>
            </div>
          </div>
        </div>
        <svg
          class="w-full h-5"
          data-testid="LineGraph__svg"
        >
          <rect
            class="fill-current text-theme-success-600"
            data-testid="LineGraph__item"
            height="8"
            rx="4"
            width="77"
            x="0"
            y="8"
          >
            <animate
              attributeName="height"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="8"
              to="16"
            />
            <animate
              attributeName="height"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="16"
              to="8"
            />
            <animate
              attributeName="rx"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="4"
              to="8"
            />
            <animate
              attributeName="rx"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="8"
              to="4"
            />
            <animate
              attributeName="y"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="8"
              to="0"
            />
            <animate
              attributeName="y"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="0"
              to="8"
            />
          </rect>
          <rect
            class="fill-current text-theme-warning-600"
            data-testid="LineGraph__item"
            height="8"
            rx="4"
            width="15"
            x="85"
            y="8"
          >
            <animate
              attributeName="height"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="8"
              to="16"
            />
            <animate
              attributeName="height"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="16"
              to="8"
            />
            <animate
              attributeName="rx"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="4"
              to="8"
            />
            <animate
              attributeName="rx"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="8"
              to="4"
            />
            <animate
              attributeName="y"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="8"
              to="0"
            />
            <animate
              attributeName="y"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="0"
              to="8"
            />
          </rect>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioBreakdown should render with dark mode = true 1`] = `
<DocumentFragment>
  <div
    class="py-4 px-6 bg-theme-secondary-100 dark:bg-black rounded-xl flex"
    data-testid="PortfolioBreakdown"
  >
    <div
      class="flex space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800"
    >
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Your Balance
        </span>
        <span
          class="whitespace-nowrap text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="Amount"
        >
          $100.00
        </span>
      </div>
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Assets
        </span>
        <span
          class="text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="PortfolioBreakdown__assets"
        >
          2
        </span>
      </div>
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Wallets
        </span>
        <span
          class="text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="PortfolioBreakdown__wallets"
        >
          2
        </span>
      </div>
    </div>
    <div
      class="flex-1 ml-6"
    >
      <div>
        <div
          class="absolute duration-200 transition-opacity opacity-0"
        />
        <div
          class="flex justify-end mb-1"
          data-testid="LineGraph__legend"
        >
          <div
            class="flex justify-end mb-1"
          >
            <div
              class="flex space-x-4"
            >
              <div
                class="flex items-center space-x-1 text-sm font-semibold"
              >
                <span
                  class="h-3 w-1 rounded bg-theme-success-600"
                />
                <span
                  class="text-theme-secondary-700 dark:text-theme-secondary-200"
                >
                  ARK
                </span>
                <span
                  class="text-theme-secondary-500 dark:text-theme-secondary-700"
                >
                  85%
                </span>
              </div>
              <div
                class="flex items-center space-x-1 text-sm font-semibold"
              >
                <span
                  class="h-3 w-1 rounded bg-theme-warning-600"
                />
                <span
                  class="text-theme-secondary-700 dark:text-theme-secondary-200"
                >
                  LSK
                </span>
                <span
                  class="text-theme-secondary-500 dark:text-theme-secondary-700"
                >
                  15%
                </span>
              </div>
            </div>
            <div
              class="border-l border-theme-secondary-300 dark:border-theme-secondary-800 ml-4 pl-4"
            >
              <button
                class="font-semibold text-sm link"
                type="button"
              >
                More Details
              </button>
            </div>
          </div>
        </div>
        <svg
          class="w-full h-5"
          data-testid="LineGraph__svg"
        >
          <rect
            class="fill-current text-theme-success-600"
            data-testid="LineGraph__item"
            height="8"
            rx="4"
            width="77"
            x="0"
            y="8"
          >
            <animate
              attributeName="height"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="8"
              to="16"
            />
            <animate
              attributeName="height"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="16"
              to="8"
            />
            <animate
              attributeName="rx"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="4"
              to="8"
            />
            <animate
              attributeName="rx"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="8"
              to="4"
            />
            <animate
              attributeName="y"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="8"
              to="0"
            />
            <animate
              attributeName="y"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="0"
              to="8"
            />
          </rect>
          <rect
            class="fill-current text-theme-warning-600"
            data-testid="LineGraph__item"
            height="8"
            rx="4"
            width="15"
            x="85"
            y="8"
          >
            <animate
              attributeName="height"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="8"
              to="16"
            />
            <animate
              attributeName="height"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="16"
              to="8"
            />
            <animate
              attributeName="rx"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="4"
              to="8"
            />
            <animate
              attributeName="rx"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="8"
              to="4"
            />
            <animate
              attributeName="y"
              begin="mouseover"
              dur="0.1s"
              fill="freeze"
              from="8"
              to="0"
            />
            <animate
              attributeName="y"
              begin="mouseleave"
              dur="0.15s"
              fill="freeze"
              from="0"
              to="8"
            />
          </rect>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioBreakdown should render zero balance state with dark mode = false 1`] = `
<DocumentFragment>
  <div
    class="py-4 px-6 bg-theme-secondary-100 dark:bg-black rounded-xl flex"
    data-testid="PortfolioBreakdown"
  >
    <div
      class="flex space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800"
    >
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Your Balance
        </span>
        <span
          class="whitespace-nowrap text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="Amount"
        >
          $0.00
        </span>
      </div>
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Assets
        </span>
        <span
          class="text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="PortfolioBreakdown__assets"
        >
          2
        </span>
      </div>
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Wallets
        </span>
        <span
          class="text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="PortfolioBreakdown__wallets"
        >
          2
        </span>
      </div>
    </div>
    <div
      class="flex-1 ml-6"
    >
      <div>
        <div
          class="absolute duration-200 transition-opacity opacity-0"
        />
        <div
          class="flex justify-end mb-1"
          data-testid="LineGraph__legend"
        >
          <div
            class="flex justify-end mb-1"
          >
            <div
              class="flex space-x-4"
            >
              <div
                class="flex items-center space-x-1 text-sm font-semibold"
              >
                <span
                  class="h-3 w-1 rounded bg-theme-secondary-300"
                />
                <span
                  class="text-theme-secondary-700 dark:text-theme-secondary-200"
                >
                  ARK
                </span>
                <span
                  class="text-theme-secondary-500 dark:text-theme-secondary-700"
                >
                  0%
                </span>
              </div>
              <div
                class="flex items-center space-x-1 text-sm font-semibold"
              >
                <span
                  class="h-3 w-1 rounded bg-theme-secondary-300"
                />
                <span
                  class="text-theme-secondary-700 dark:text-theme-secondary-200"
                >
                  LSK
                </span>
                <span
                  class="text-theme-secondary-500 dark:text-theme-secondary-700"
                >
                  0%
                </span>
              </div>
            </div>
            <div
              class="border-l border-theme-secondary-300 dark:border-theme-secondary-800 ml-4 pl-4"
            >
              <button
                class="font-semibold text-sm cursor-not-allowed text-theme-secondary-500 dark:text-theme-secondary-700"
                disabled=""
                type="button"
              >
                More Details
              </button>
            </div>
          </div>
        </div>
        <svg
          class="w-full h-5"
          data-testid="LineGraph__svg"
        >
          <rect
            class="fill-current text-theme-secondary-300 dark:text-theme-secondary-800"
            data-testid="LineGraph__empty"
            height="8"
            rx="4"
            width="100"
            x="0"
            y="8"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioBreakdown should render zero balance state with dark mode = true 1`] = `
<DocumentFragment>
  <div
    class="py-4 px-6 bg-theme-secondary-100 dark:bg-black rounded-xl flex"
    data-testid="PortfolioBreakdown"
  >
    <div
      class="flex space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800"
    >
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Your Balance
        </span>
        <span
          class="whitespace-nowrap text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="Amount"
        >
          $0.00
        </span>
      </div>
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Assets
        </span>
        <span
          class="text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="PortfolioBreakdown__assets"
        >
          2
        </span>
      </div>
      <div
        class="flex flex-col font-semibold space-y-1 pl-3 first:pl-0"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 whitespace-nowrap"
        >
          Wallets
        </span>
        <span
          class="text-lg text-theme-secondary-900 dark:text-theme-secondary-200"
          data-testid="PortfolioBreakdown__wallets"
        >
          2
        </span>
      </div>
    </div>
    <div
      class="flex-1 ml-6"
    >
      <div>
        <div
          class="absolute duration-200 transition-opacity opacity-0"
        />
        <div
          class="flex justify-end mb-1"
          data-testid="LineGraph__legend"
        >
          <div
            class="flex justify-end mb-1"
          >
            <div
              class="flex space-x-4"
            >
              <div
                class="flex items-center space-x-1 text-sm font-semibold"
              >
                <span
                  class="h-3 w-1 rounded bg-theme-secondary-800"
                />
                <span
                  class="text-theme-secondary-700 dark:text-theme-secondary-200"
                >
                  ARK
                </span>
                <span
                  class="text-theme-secondary-500 dark:text-theme-secondary-700"
                >
                  0%
                </span>
              </div>
              <div
                class="flex items-center space-x-1 text-sm font-semibold"
              >
                <span
                  class="h-3 w-1 rounded bg-theme-secondary-800"
                />
                <span
                  class="text-theme-secondary-700 dark:text-theme-secondary-200"
                >
                  LSK
                </span>
                <span
                  class="text-theme-secondary-500 dark:text-theme-secondary-700"
                >
                  0%
                </span>
              </div>
            </div>
            <div
              class="border-l border-theme-secondary-300 dark:border-theme-secondary-800 ml-4 pl-4"
            >
              <button
                class="font-semibold text-sm cursor-not-allowed text-theme-secondary-500 dark:text-theme-secondary-700"
                disabled=""
                type="button"
              >
                More Details
              </button>
            </div>
          </div>
        </div>
        <svg
          class="w-full h-5"
          data-testid="LineGraph__svg"
        >
          <rect
            class="fill-current text-theme-secondary-300 dark:text-theme-secondary-800"
            data-testid="LineGraph__empty"
            height="8"
            rx="4"
            width="100"
            x="0"
            y="8"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
